### PR TITLE
Screen shot Optimization

### PR DESCRIPTION
**Pull request by prashelke**
Screen shot Optimization for all driver is done.
 
Thank you for your contribution.
Before submitting this PR, please make sure:
- [x] PR description and commit message should describe the changes done in this PR
- [x] Verify the PR is pointing to correct branch i.e. Release or Beta branch if the code fix is for specific release , else point it to master
- [x] Latest Code from master or specific release branch is merged to your branch
- [x] No unwanted\commented\junk code is included
- [x] No new warning upon build solution
- [x] Code Summary\Comments are added to my code which explains what my code is doing
- [ ] Existing unit test cases are passed
- [ ] New Unit tests are added for your development
- [ ] Sanity Tests are successfully executed for New and Existing Functionality
- [ ] Verify that changes are compatible with all relevant browsers and platforms.
- [x] After creating pull request there should not be any conflicts
- [ ] Resolve codacy comments
- [ ] Builds and checks are passed before PR is sent for review
- [ ] Resolve code review comments
- [ ] Help Libray document is updated accordingly for Feature changes.

## Summary by CodeRabbit

- **Refactor**
  - Enhanced resource management across various components with proper disposal of screenshot objects to prevent memory leaks.
  - Simplified methods for taking and adding screenshots for better maintainability.

- **Bug Fixes**
  - Fixed potential memory management issues by introducing `using` statements for bitmap objects.

### CODE REVIEW

File: Ginger/Ginger/AutomatePageLib/AddActionMenu/WindowExplorer/Common/WindowExplorerPage.xaml.cs [hunk 1]
```diff
@@ -1120,7 +1120,7 @@ public void ShowScreenShot()
             /// UnComment later after complete Implementation of functionalities over all platforms.
             //if(IsWebMobJavaPlatform)
             mWindowExplorerDriver.UnHighLightElements();
-
+            
             try
             {
                 LoadPageSourceDoc = mWindowExplorerDriver.SupportedViews().Contains(eTabView.PageSource);
```
[the remark this answers is not in the document]
Comment (prashelke) [reply Ginger/Ginger/AutomatePageLib/AddActionMenu/WindowExplorer/Common/WindowExplorerPage.xaml.cs:1126]: Done

File: Ginger/Ginger/AutomatePageLib/AddActionMenu/WindowExplorer/Common/WindowExplorerPage.xaml.cs [hunk 2]
```diff
@@ -1130,22 +1130,24 @@ public void ShowScreenShot()
...
-                //if (mPlatform.PlatformType() == ePlatformType.Web)
-                //{
-                //    mScreenShotViewPage.xMainImage.MouseMove += XMainImage_MouseMove;
-                //}
+                    mScreenShotViewPage.ImageMouseCursor = Cursors.Hand;
+                    /// UnComment to allow Element detection on hover
+                    //if (mPlatform.PlatformType() == ePlatformType.Web)
+                    //{
+                    //    mScreenShotViewPage.xMainImage.MouseMove += XMainImage_MouseMove;
+                    //}
```
[the remark this answers is not in the document]
Comment (prashelke) [reply Ginger/Ginger/AutomatePageLib/AddActionMenu/WindowExplorer/Common/WindowExplorerPage.xaml.cs:1142]: Done

File: Ginger/GingerCoreNET/ActionsLib/UI/VisualTesting/UIElementsAnalyzer.cs [hunk 3]
```diff
@@ -56,36 +56,38 @@ void IVisualAnalyzer.Compare()
...
-                //TODO: if output each mismatch then do output
+                    //TODO: if output each mismatch then do output
 
 
-                mAct.AddScreenShot(bmp, "Compare Result");
+                    mAct.AddScreenShot(bmp, "Compare Result");
+                    //TODO: add small bitmap of mismatch elem
 
-                //TODO: add small bitmap of mismatch elem
+                }
```
[the remark this answers is not in the document]
Comment (prashelke) [reply Ginger/GingerCoreNET/ActionsLib/UI/VisualTesting/UIElementsAnalyzer.cs:90]: ok

File: Ginger/GingerCoreNET/Drivers/CoreDrivers/Web/Selenium/SeleniumDriver.cs [hunk 4]
```diff
@@ -8785,6 +8805,46 @@ public Bitmap GetScreenShot(bool IsFullPageScreenshot = false)
...
+                    }
+                    break;
+                default:
+                    using (Bitmap bitmapImage = CaptureFullPageScreenshot())
+                    {
+                        act.AddScreenShot(bitmapImage, Driver.Title);
+                    }
+                    break;
+            }
+        }
```
[the remark this answers is not in the document]
Comment (prashelke) [reply Ginger/GingerCoreNET/Drivers/CoreDrivers/Web/Selenium/SeleniumDriver.cs:8846]: this is specific Scenario  let's keep it that way for now

File: Ginger/GingerCoreNET/Drivers/CoreDrivers/Web/Selenium/SeleniumDriver.cs [hunk 5]
```diff
@@ -1579,21 +1578,42 @@ private void ScreenshotHandler(ActScreenShot act)
...
+                    
+                string filepath = TargetFrameworkHelper.Helper.MergeVerticallyAndSaveBitmaps(bitmapsToMerge.ToArray());
+
+                act.ScreenShotsNames.Add(Driver.Title);
+                act.ScreenShots.Add(filepath);   
+            }
+            finally
+            {
+                bitmapsToMerge.Clear();
+            }
```
[the remark this answers is not in the document]
Comment (prashelke) [reply Ginger/GingerCoreNET/Drivers/CoreDrivers/Web/Selenium/SeleniumDriver.cs:1616]: Done

File: Ginger/GingerCoreNET/Drivers/CoreDrivers/Mobile/Appium/GenericAppiumDriver.cs [hunk 6]
```diff
@@ -1094,14 +1094,26 @@ public bool isValidPhotoExtention(string photo)
...
+                }
+            }
+            finally
             {
-                picture.Save(ms, format);
-                Byte[] bytes = ms.ToArray();
-                encodeString = Convert.ToBase64String(bytes);
+                Array.Clear(bytes);
             }
+
```
[the remark this answers is not in the document]
Comment (prashelke) [reply Ginger/GingerCoreNET/Drivers/CoreDrivers/Mobile/Appium/GenericAppiumDriver.cs]: Done

**Review by Maheshkale447 — APPROVED** [review]

**Pull request merged**